static/js/settings.js

```diff
@@ -292,6 +292,10 @@ function _setup_page() {
             page_params.enable_online_push_notifications = result.enable_online_push_notifications;
         }
 
+        if (result.hide_private_message_desktop_notifications !== undefined) {
+            page_params.hide_private_message_desktop_notifications
+                = result.hide_private_message_desktop_notifications;
+        }
```

Comment:
Hmm, I think it's actually possible that all the `page_params` update logic in this section of the function can be deleted, since I think we're now doing it inside `exports.handle_global_notification_updates` in `static/js/notifications.js`, which is the correct place to do these updates.

I'm going to look into doing so now.